code/_onclick/hud/robot.dm

```diff
@@ -98,6 +98,13 @@
 	mymob.throw_icon.icon_state = "store"
 	mymob.throw_icon.name = "store"
 	mymob.throw_icon.screen_loc = ui_borg_store
+	
+//Headlamp
+	mymob:lamp_button = new /obj/screen()
```

Comment:
Because our HUD code is absolutely ancient, especially compared to /tg/. It doesn't actually typecast anything properly, so if you add a hud element that is dynamic (changes when clicked) and is not shared among ALL mobs (such as intents, throw, etc.) you have to colon it (see the "cells" hud element in that file for another example, that's the cell monitor on the top right of the hud).